### PR TITLE
[MID-18] add husky & linter on commit

### PR DESCRIPTION
**Pull request by alex-popov-tech**
https://trello.com/c/Ypl2tDAS/18-%D0%BD%D0%B0%D0%BB%D0%B0%D1%88%D1%82%D1%83%D0%B2%D0%B0%D0%BD%D0%BD%D1%8F-%D0%BB%D1%96%D0%BD%D1%82%D0%B5%D1%80%D1%83

### CODE REVIEW

**Comment by Yivgha** [comment]
Good job. But change file .eslintrc.json to .eslintrc.js according to trello task.

**Review by IvanKoshan — COMMENTED** [review]
I've tested this setup on a test branch. In case the linter gives a warning, the commit is allowed. Should this be allowed?
image

**Comment by alex-popov-tech** [comment]
@IvanKoshan yep, its a warning, not an error

File: package.json [hunk 1]
```diff
@@ -32,6 +33,11 @@
     "@types/react-dom": "~18.2.18",
     "eslint": "^8",
     "eslint-config-next": "14.1.0",
+    "eslint-config-prettier": "^9.1.0",
+    "husky": "^9.0.7",
+    "lint-staged": "^15.2.1",
+    "prettier": "3.2.4",
+    "prettier-plugin-prisma": "^5.0.0",
```
Comment (BohdanLeshchyshyn) [inline package.json:40]: I believe will be better to put linting dependencies to the devDependencies chapter.

**Comment by BohdanLeshchyshyn** [comment]
I don't fully understand how all this stuff works, could you add some info about hooks and code style to readme?

**Comment by BohdanLeshchyshyn** [comment]
Sorry if it was not clarified in the task.
But I believe that we need to add as many linting restrictions as possible.
So, will be better to take some default config a-la airbnb.
You even can copy and paste most settings from https://github.dev/Sportresurs/sportresurs

To understand - I want to show you what could be handled by linter,  but after run `npm run lint` it said that everything is okay.

On the screenshot below I expect to get the following errors:
- different types of quotes at import
- internal imports should be placed only after all external imports
- too many empty lines
- usage of var instead of const or let
- unused variables
- different types of quotes at strings (usually single quotes or backticks for template literals)
- unused argument in function
- missing semicolon
- missing space before/after curly brace
- too long line (has to be 100 or 120 characters)
- different indentation (usually 2 or 4 spaces) - but in the current project has to be 2 spaces

I believe that all these errors can be from the box airbnb config.

The second screenshot is an example of what a developer should see during development (of course if also has installed the linter extension at IDE)
Screenshot 2024-01-31 at 19 32 41
Screenshot 2024-01-31 at 19 37 31

File: package.json [hunk 2]
```diff
@@ -27,11 +29,22 @@
...
     "@types/bcryptjs": "~2.4.6",
     "@types/eslint": "~8.56.2",
+    "@types/eslint-config-prettier": "~6.11.3",
+    "@types/lint-staged": "~13.3.0",
     "@types/pg": "~8.11.0",
     "@types/react": "~18.2.48",
     "@types/react-dom": "~18.2.18",
-    "eslint": "^8",
+    "eslint": "^8.56.0",
+    "eslint-config-airbnb": "^19.0.4",
```
Comment (BohdanLeshchyshyn) [inline package.json]: please, use this package - https://www.npmjs.com/package/eslint-config-airbnb-base

**Pull request merged**